packages/typespec-ts/test/modularIntegration/generated/client/structure/default/src/api/operations.ts

```diff
@@ -34,7 +35,13 @@ export function _oneSend(
 
 export async function _oneDeserialize(result: One204Response): Promise<void> {
   if (result.status !== "204") {
-    throw result.body;
+    const internalError = (result.body as any).error || result.body || result;
+    const message = `Unexpected status code ${result.status}`;
+    throw new RestError(internalError.message ?? message, {
+      statusCode: result.status,
```

Comment:
I think status code is the actual string code field in the body and not the status of the response.

Reply:
We probably need to try to set it to `result.body.error.code ?? result.body.code`.